### PR TITLE
Hw3 dev

### PR DESCRIPTION
**Pull request by ZhenruWang**

### CODE REVIEW

**Comment by feiyu-chen96** [comment]
P2E(5/5) Good job.
Other comments: Implementation of score(): ** is elementwise and vectorized for numpy.array so we don't need list comprehension here.

File: homeworks/HW3-final/P1B.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+from Regression import Regression as reg
```
Comment (bhavenp) [inline homeworks/HW3-final/P1B.py:1]: 5/5 for Problem 2B.

**Comment by adityakaran** [comment]
"2a: 6/10 - Should raise error instead of return None
2c: 10/10 - Nice job!"

**Comment by lindseysbrown** [comment]
Your directory structure is wrong.  It should be homeworks/HW3/HW3-final

File: homeworks/HW3-final/model_performance.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+                                                    dataset['target'], 
+                                                    test_size=0.2, 
+                                                    random_state=42)
+
+
+alpha_list = np.arange(0.05,1,0.09)
+score_list = []
+for a in alpha_list:
+    #model = reg.RidgeRegression(alpha)
+    model = reg.RidgeRegression(0.1)
```
Comment (lindseysbrown) [inline homeworks/HW3-final/model_performance.py:19]: You don't want to do this in the for loop - you're creating lots of unnecessary RidgeRegression objects.

File: homeworks/HW3-final/model_performance.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    model.fit(X_train, y_train)
+    y_predict = model.predict(X_test)
+    score = model.score(X_test,y_test)
+    score_list.append(score)
+    
+plt.plot(alpha_list,score_list,label = 'Ridge Regression')
+
+score_list_l = []
+for alpha in alpha_list:
+    model = reg.LinearRegression()
```
Comment (lindseysbrown) [inline homeworks/HW3-final/model_performance.py:31]: Same comment as above.

P2F(5/5) Good job.

**Comment by cridonoso** [comment]
**Problem 0:** _(total score: 10 )_ 
Working in branch HW3-dev
HW2 was merged
PR was generated


**Problem 1:** _(total score: 5)_ 
good job!

**Directory tree**: _(course policy)_ 
HW1/HW1-final and HW2/HW2-final folders not found


Regards,
Cristóbal

**Comment by daninge** [comment]
Couldn't locate Bank.py 0/30

**Comment by cj-xin** [comment]
### Problem 2D

1/1 pt: Inheritance correct.
1/2 pt: Should use `super` to initialise `self.params`; `set_params` implementation will throw `KeyError` if it is not given an `alpha` value
7/7 pt: Fit is correct

Comment: Comment your code

File: homeworks/HW3-final/Bank.py [hunk 4]
```diff
@@ -0,0 +1,173 @@
+from enum import Enum
```
Comment (daninge) [inline homeworks/HW3-final/Bank.py:1]: -2 point: Savings account hasnt been created yet but getbalance() does not raise Exception,
-1 point: Savings account hasnt been created yet but withdraw does not raise an Exception.,
-1 point: Savings account hasnt been created yet but deposit does not raise an Exception.,
-6 points: ATMSession did not update CHECKING account properly.

**Pull request merged**